contrib/aws/aws-sdk-go-v2/aws/aws.go

```diff
@@ -82,12 +100,19 @@ func (mw *traceMiddleware) startTraceMiddleware(stack *middleware.Stack) error {
 			tracer.ServiceName(serviceName(mw.cfg, serviceID)),
 			tracer.ResourceName(fmt.Sprintf("%s.%s", serviceID, operation)),
 			tracer.Tag(tagAWSRegion, awsmiddleware.GetRegion(ctx)),
+			tracer.Tag(tagTopLevelRegion, awsmiddleware.GetRegion(ctx)),
 			tracer.Tag(tagAWSOperation, operation),
 			tracer.Tag(tagAWSService, serviceID),
+			tracer.Tag(tagTopLevelAWSService, serviceID),
 			tracer.StartTime(ctx.Value(spanTimestampKey{}).(time.Time)),
 			tracer.Tag(ext.Component, componentName),
 			tracer.Tag(ext.SpanKind, ext.SpanKindClient),
 		}
+		resourceNameKey, resourceNameValue, err := extractResourceNameFromParams(in, serviceID)
+		if err != nil {
+			log.Printf("Error: %v", err)
+		}
+		opts = append(opts, tracer.Tag(resourceNameKey, resourceNameValue))
```

Comment:
this should be in an else with the previous if-statement right? Typically in Go you don't use the output of a function if there was an error, so I would expect this tag to only be added if the previous err is nil.